target/TARGET_NXP/sdio_device.c

```diff
@@ -0,0 +1,188 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "fsl_sd.h"
+#include "pinmap.h"
+#include "sdio_device.h"
+
+/* Array of SD peripheral base address. */
+static SDIF_Type *const sd_addrs[] = SDIF_BASE_PTRS;
+static sd_card_t g_sd;
+
+extern void sdio_clock_setup(void);
+
+/**
+ * @brief  Initializes the SD card device.
+ * @retval SD status
+ */
+uint8_t SDIO_Init(void)
+{
+    uint32_t reg;
+
+    /*! @brief SDMMC host detect card configuration */
+    sdmmchost_detect_card_t s_sdCardDetect = {
+        .cdType = kSDMMCHOST_DetectCardByGpioCD,
+        .cdTimeOut_ms = (~0U),
+    };
+
+    sdio_clock_setup();
+
+    /* SD POW_EN */
+    pin_function(P0_9, 2);
+    pin_mode(P0_9, PullNone);
+
+    /* SD DAT3 */
+    pin_function(P1_0, 2);
+    pin_mode(P1_0, PullNone);
+    reg = IOCON->PIO[1][0];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[1][0] = reg;
+
+    /* SD DAT2 */
+    pin_function(P0_31, 2);
+    pin_mode(P0_31, PullNone);
+    reg = IOCON->PIO[0][31];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[0][31] = reg;
+
+    /* SD DAT1 */
+    pin_function(P0_25, 2);
+    pin_mode(P0_25, PullNone);
+    reg = IOCON->PIO[0][25];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[0][25] = reg;
+
+    /* SD DAT0 */
+    pin_function(P0_24, 2);
+    pin_mode(P0_24, PullNone);
+    reg = IOCON->PIO[0][24];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[0][24] = reg;
+
+    /* SD CLK */
+    pin_function(P0_7, 2);
+    pin_mode(P0_7, PullNone);
+    reg = IOCON->PIO[0][7];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[0][7] = reg;
+
+    /* SD CMD */
+    pin_function(P0_8, 2);
+    pin_mode(P0_8, PullNone);
+    reg = IOCON->PIO[0][8];
+    reg |= IOCON_PIO_SLEW_MASK;
+    IOCON->PIO[0][8] = reg;
+
+    /* SD Card Det */
+    //pin_function(P0_17, 2);
+    //pin_mode(P0_17, PullNone);
+
+    /* SD Write Protect */
+    //pin_function(P0_15, 5);
```

Comment:
what about these unused code, lines 88-94 ?

Reply:
Thank you for catching this. This has been fixed.